### PR TITLE
OvenMode example Implementation

### LINKED ISSUE #31091: [1.3] The implementation of the Oven Mode Cluster is required
### Reproduction steps

1. Executed the below mentioned command to put DUT into a commissionable state
**./chip-all-clusters-app**

2. Once DUT reach the commissionable state sent below mentioned command on TH.
**./chip-tool pairing onnetwork 1 20202021**

3. After the successful commissioning, I attempted to validate the Oven Mode Cluster, but it is not implemented in Reference app(All-Cluster-app)
**./chip-tool ovenmode read cluster-revision 1 1**

```
[1702961164.756988][4463:4465] CHIP:DMG: ReportDataMessage =
[1702961164.757000][4463:4465] CHIP:DMG: {
[1702961164.757007][4463:4465] CHIP:DMG: 	AttributeReportIBs =
[1702961164.757024][4463:4465] CHIP:DMG: 	[
[1702961164.757031][4463:4465] CHIP:DMG: 		AttributeReportIB =
[1702961164.757047][4463:4465] CHIP:DMG: 		{
[1702961164.757057][4463:4465] CHIP:DMG: 			AttributeStatusIB =
[1702961164.757072][4463:4465] CHIP:DMG: 			{
[1702961164.757082][4463:4465] CHIP:DMG: 				AttributePathIB =
[1702961164.757095][4463:4465] CHIP:DMG: 				{
[1702961164.757108][4463:4465] CHIP:DMG: 					Endpoint = 0x1,
[1702961164.757121][4463:4465] CHIP:DMG: 					Cluster = 0x49,
[1702961164.757137][4463:4465] CHIP:DMG: 					Attribute = 0x0000_FFFD,
[1702961164.757149][4463:4465] CHIP:DMG: 				}
[1702961164.757167][4463:4465] CHIP:DMG: 					
[1702961164.757183][4463:4465] CHIP:DMG: 				StatusIB =
[1702961164.757197][4463:4465] CHIP:DMG: 				{
[1702961164.757209][4463:4465] CHIP:DMG: 					status = 0xc3 (UNSUPPORTED_CLUSTER),
[1702961164.757220][4463:4465] CHIP:DMG: 				},
[1702961164.757235][4463:4465] CHIP:DMG: 				
[1702961164.757245][4463:4465] CHIP:DMG: 			},
[1702961164.757265][4463:4465] CHIP:DMG: 	
```		

### Bug prevalence

Whenever I do

### GitHub hash of the SDK that was being used

7cbac9130d2b10d6bfdba873129dfbffa294cf8d

### Platform

raspi

### Platform Version(s)

1.3

### Anything else?

**All-clusters-app has to be enabled with below mentioned Attribute:**

- [ ] SupportedModes
- [ ] CurrentMode
- [ ] StartUpMode
- [ ] OnMode
- [ ] ClusterRevision
- [ ] FeatureMap
- [ ] AttributeList
- [ ] AcceptedCommandList
- [ ] GeneratedCommandList

**All-clusters-app has to be enabled with below mentioned Command :**

- [ ] ChangeToMode
- [ ] ChangeToModeResponse

### PR DESCRIPTION
**Pull request by eahove**
Implementing the ovenMode example for linux and ESP32

Addresses issue:https://github.com/project-chip/connectedhomeip/issues/31091

### CODE REVIEW

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter [hunk 1]
```diff
@@ -7060,9 +7094,9 @@ endpoint 1 {
     ram      attribute clusterRevision default = 1;
   }
 
-  server cluster LaundryDryerControls {
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:7063]: Why is this turning off LaundryDryerControls?

File: src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp [hunk 2]
```diff
@@ -43,7 +43,7 @@ using namespace chip::app::Clusters::LaundryDryerControls::Attributes;
 using chip::Protocols::InteractionModel::Status;
 
 static constexpr size_t kLaundryDryerControlsDelegateTableSize =
-    EMBER_AF_LAUNDRY_DRYER_CONTROLS_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+    EMBER_AF_LAUNDRY_WASHER_CONTROLS_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
```
Comment (bzbarsky-apple) [inline src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp:46]: Why these changes to laundry dryer controls?  These don't see, right...

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter (hunk 1 above)
Comment (eahove) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:7063]: Bad merge, I assume. Reverted.

File: src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp (hunk 2 above)
Comment (eahove) [reply src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp]: Reverted these changes after fixing the zap file.

**Review by plauric — APPROVED** [review]
